components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaDevConsole.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    @Override
+    protected String doCallText(Map<String, Object> options) {
+        final boolean committed = "true".equals(options.getOrDefault(COMMITTED, "false"));
+
+        StringBuilder sb = new StringBuilder();
+        for (Route route : getCamelContext().getRoutes()) {
+            if (route.getConsumer() instanceof KafkaConsumer kc) {
+                sb.append(String.format("\n    Route Id: %s", route.getRouteId()));
+                sb.append(String.format("\n    From: %s", route.getEndpoint().getEndpointUri()));
+                for (KafkaFetchRecords t : kc.tasks()) {
+                    sb.append(String.format("\n        Worked Thread: %s", t.getThreadId()));
+                    sb.append(String.format("\n        Worker State: %s", t.getState()));
+                    TaskHealthState hs = t.healthState();
+                    if (!hs.isReady()) {
+                        sb.append(String.format("\n        Worker Last Error: %s", hs.buildStateMessage()));
+                    }
+                    KafkaFetchRecords.GroupMetadata meta = t.getGroupMetadata();
+                    if (meta != null) {
+                        sb.append(String.format("\n        Group Id: %s", meta.groupId()));
+                        sb.append(String.format("\n        Group Instance Id: %s", meta.groupInstanceId()));
+                        sb.append(String.format("\n        Member Id: %s", meta.memberId()));
+                        sb.append(String.format("\n        Generation Id: %d", meta.generationId()));
+                    }
+                    if (t.getLastRecord() != null) {
+                        sb.append(String.format("\n        Last Topic: %s", t.getLastRecord().topic()));
+                        sb.append(String.format("\n        Last Partition: %d", t.getLastRecord().partition()));
+                        sb.append(String.format("\n        Last Offset: %d", t.getLastRecord().offset()));
+                    }
+                    if (committed) {
+                        List<KafkaFetchRecords.KafkaTopicPosition> l = fetchCommitOffsets(kc, t);
+                        if (l != null) {
+                            for (KafkaFetchRecords.KafkaTopicPosition r : l) {
+                                sb.append(String.format("\n        Commit Topic: %s", r.topic()));
+                                sb.append(String.format("\n        Commit Partition: %s", r.partition()));
+                                sb.append(String.format("\n        Commit Offset: %s", r.offset()));
+                                if (r.epoch() > 0) {
+                                    long delta = System.currentTimeMillis() - r.epoch();
+                                    sb.append(String.format("\n        Commit Offset Since: %s",
+                                            TimeUtils.printDuration(delta, true)));
+                                }
+                            }
+                        }
+                    }
```

Comment:
IMHO, this would be much better done by using a script language such as Velocity. Formatting the strings builtin are terribly hard to maintain.

Reply:
No this is how the dev consoles are already done